react-client/src/features/disclosures/page.tsx

```diff
@@ -4,199 +4,212 @@ import { DataTableToolbar } from "@/features/disclosures/data-table-toolbar";
+
+		const params = new URLSearchParams(window.location.search);
+
+		if (visaFilters?.length) {
+			params.set("visa", visaFilters.join(","));
+		} else {
+			params.delete("visa");
+		}
+
+		if (statusFilters?.length) {
+			params.set("status", statusFilters.join(","));
+		} else {
+			params.delete("status");
+		}
+
+		window.history.replaceState(
+			{},
+			"",
+			`${window.location.pathname}?${params}`
+		);
+	}, [columnFilters]);
+
+	const filters: InputMaybe<LcaDisclosureFilters> = useMemo(
+		() => ({
+			visaClass: getVisaFilters(columnFilters),
+			caseStatus: getCaseStatusFilters(columnFilters),
+			employerUuid: getEmployerUuidsFilters(columnFilters),
+			jobTitle: getJobTitleFilters(columnFilters),
+		}),
+		[columnFilters]
+	);
+
+	const queryTake = pagination.pageSize * 3;
+
+	const queryVariables: PaginatedLcaDisclosuresQueryVariables = {
+		pagination: {
+			take: queryTake,
+			skip: pagination.pageIndex * queryTake,
+		},
+		filters,
+		sorting: sortingInput,
+	};
+
+	const { loading, data } = useQuery(PaginatedLcaDisclosuresDocument, {
+		variables: queryVariables,
+	});
+	const { items, totalCount: dataCount } = data?.lcaDisclosures || {};
+
+	const [totalCount, setTotalCount] = React.useState<number | undefined>(
+		undefined
+	);
+
+	React.useEffect(() => {
+		if (!!dataCount && totalCount !== dataCount) {
+			setTotalCount(dataCount);
+		}
+	}, [dataCount, totalCount]);
+
+	const [loadedData, setLoadedData] = React.useState<LCADisclosure[]>([]);
+	const loadedDataIDsSet = React.useMemo(() => {
+		return new Set(loadedData.map((d) => d.caseNumber));
+	}, [loadedData]);
+
+	React.useEffect(() => {
+		if (items) {
+			const newItems = items.filter((i) => !loadedDataIDsSet.has(i.caseNumber));
+			if (newItems.length > 0) {
+				setLoadedData([...loadedData, ...newItems]);
+			}
+		}
+	}, [items, loadedData, loadedDataIDsSet]);
+
+	/**
+	 * Clear loaded data if filters or sorting change
+	 */
+	React.useEffect(() => {
+		setLoadedData([]);
+	}, [filters, sortingInput]);
+
+	const toolbarComponent = React.useCallback(
+		(props: { table: Table<LCADisclosure> }) => (
+			<DataTableToolbar table={props.table} queryFilters={filters} />
+		),
+		[filters]
+	);
+
+	return (
+		<div className="h-full flex-1 flex-col space-y-8 md:p-8 flex">
+			<div className="flex items-center justify-between space-y-2 px-8 pt-4 md:px-0 md:pt-0">
+				<div>
+					<h2 className="text-2xl font-bold tracking-tight">
+						Explore USA visa data
+					</h2>
+					<p className="text-muted-foreground">
+						{totalCount === undefined ? (
+							<Skeleton className="h-4 w-[60px] inline-block" />
+						) : (
+							`${totalCount}`
+						)}
+						<span>{" visa applications from the U.S Department of Labor"}</span>
+					</p>
+					<a
+						href="https://h1b1.notion.site"
+						target="_blank"
+						rel="noopener noreferrer"
+					>
+						<Button className="mt-4">How does H1B1 work? 🔗</Button>
+					</a>
```

Comment:
These are all the changes I am actually making. The rest are just prettify formatting